packages/SwingSet/src/kernel/kernel.js

```diff
@@ -601,6 +607,8 @@ export default function buildKernel(
     if (!vatWarehouse.lookup(vatID)) {
       return NO_DELIVERY_CRANK_RESULTS; // can't collect from the dead
     }
+    const vatKeeper = kernelKeeper.provideVatKeeper(vatID);
+    vatKeeper.addDirt({ gcKrefs: krefs.length });
```

Comment:
I think that would work. It would look like
```js
function deliveryCrankResults(vatID, status, measureDirt, meterID, gcKrefs) {
  ...
  if (measureDirt && ..) {
    if (gcKrefs) {
      dirt.gcKrefs = gcKrefs;
    }
 ..
}

...

  deliveryCrankResults(vatID, status, true, undefined, krefs.length);
```

IIRC I balked at adding another positional argument to `deliveryCrankResults()` which would only be used by `processGCMessage()` but would need to be provided by all callers, especially since we've already got the optional `meterID` posarg, and stopped short of converting the posargs to an options bag.

I think the abort/terminate shouldn't be a big influence.. I don't know exactly whether we should increment the dirt counter if the delivery caused a vat-fatal error or not, but I think either way would be ok.

I'll look to see if there's a decent way to refactor it as you suggest.

Reply:
Oh yeah definitely should avoid more positional args. That's what named args (destructured records are for).

Btw, the above was a light suggestion, I'm really not sure if it makes sense. I just had some trouble following the flow of where dirt gets changed.

Reply:
done